csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,192 @@
+"""
+Inspect Dataset Manager
+-----------------------
+
+Collection of functions handling csv-related operations used in the inspect cli.
+"""
+
+
+import logging
+from pathlib import Path
+from typing import List, Optional
+import pandas as pd
+
+from csvcubed.utils.pandas import read_csv
+from csvcubed.models.inspectsparqlresults import (
+    QubeComponentResult,
+)
+from csvcubed.utils.qb.components import (
+    ComponentField,
+    ComponentPropertyAttributeURI,
+    ComponentPropertyType,
+    get_component_property_as_relative_path,
+)
+from csvcubed.models.inspectdataframeresults import (
+    DatasetObservationsByMeasureUnitInfoResult,
+    DatasetObservationsInfoResult,
+    DatasetSingleMeasureResult,
+)
+from csvcubed.models.csvcubedexception import (
+    CsvToDataFrameLoadFailedException,
+    InvalidNumberOfRecordsException,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+def _filter_components_from_dsd(
+    components: List[QubeComponentResult],
+    field: ComponentField,
+    filter: str,
+) -> List[QubeComponentResult]:
+    """
+    Filters the components for the given filter.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `List[QubeComponentResult]` - filtered results.
+    """
+
+    return [
+        component
+        for component in components
+        if component.as_dict()[field.value] == filter
+    ]
+
+
+def load_csv_to_dataframe(json_path: Path, csv_path: Path) -> pd.DataFrame:
+    """
+    Loads the csv in given path to a Panda Dataframe.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DataFrame` - Dataframe of the csv.
+    """
+
+    try:
+        dataset_path = json_path.parent / csv_path
+        _logger.debug(f"Dataset path: {dataset_path.absolute()}")
+
+        dataset = pd.DataFrame(read_csv(dataset_path))
```

Comment:
The `read_csv` function returns a dataframe, we shouldn't need to convert it again.